master/buildbot/www/hooks/github.py

```diff
@@ -263,12 +263,15 @@ def _process_change(self, payload, user, repo, repo_url, project, event,
             log.msg("Branch `{}' deleted, ignoring".format(branch))
             return changes
 
+        if not payload['head_commit'] and not payload['commits']:
```

Comment:
I'm not sure if "head_commit" should be non null in [this test case](https://github.com/buildbot/buildbot/blob/adc9936c43dee46a640f493c8ffcda567ca9491d/master/buildbot/test/unit/test_www_hooks_github.py#L447), but this check will handle it properly

Reply:
We need to add a new test case for this event. I am not sure exactly which kind of event this is about (a forced push which revert a branch to a previous commit?)